### ISSUE TITLE
bug: getUrl Command throws `Method is not implemented` Error

### ISSUE THREAD

**Issue by Dor-bl** [appium/appium-inspector#1647]
### Is this an issue specifically with Appium Inspector?

- [X] I have verified that my issue does _not_ occur with Appium, and should be investigated as an Appium Inspector issue

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

When Executing the `getUrl` command an error is thrown: `method is not implemented` 

![image](https://github.com/user-attachments/assets/9ef5afce-a73d-40c6-ad9c-5a695824ef50)

### Expected Behavior

If Method is indeed not implemented, this Command should be removed. 

### Operating System

Mac

### Appium Inspector Version

2024.8.2

### Appium Version

2.4.1

### Further Information

Appium log:
https://gist.github.com/Dor-bl/98f27d059d69c38925fba247b17eba1c

**Comment by eglitise** [comment]
I'm not fully sure of the origins of this command, but given that it's under the Web section, I think it might only work in web context. In your screenshot I can see that you are using hybrid mode, but the yellow icon indicates that no web contexts were found, so the driver is still in native mode ([check the documentation for the context switcher](https://appium.github.io/appium-inspector/latest/session-inspector/header/#context-switcher)). At the same time, you have Safari open, so a webview should definitely be available.

I would suggest perhaps refreshing the source/screenshot until the Inspector shows the context dropdown, then switching to the web context and trying out the command again.

Side-note, I agree that the commands panel is quite outdated and the available commands list is long overdue for an update, at the very least to remove the deprecated/removed commands.

**Comment by Dor-bl** [comment]
@eglitise,
I've checked on the APIDemo App and the same result:
![image](https://github.com/user-attachments/assets/83d96b20-d7b6-4496-9cac-f257b9ae8f90)

Using latest Appium version and latest uiautomator2 version

**Comment by eglitise** [comment]
I'm not really sure. The error is returned by the driver, not the inspector, so I guess the issue is that the Inspector does not hide actions that are not supported by the currently active driver. I did previously configure some actions to be conditionally shown, but I'm sure there are more like that.

**Comment by jlipps** [comment]
Yeah, the inspector isn't necessarily designed to customize its interface depending on the driver that's used.

If we were to do something like that, I would want the driver to report what commands it supports (including e.g. `mobile: ` commands), and have the inspector dynamically generate what is available based on that, rather than hardcoding a list per platform in the inspector.

**Comment by Dor-bl** [comment]
@KazuCocoa / @mykola-mokhnach, was the `getUrl` command recently deprecated?
Are there any other alternatives?

**Comment by KazuCocoa** [comment]
For NATIVE_APP? I think native app context doesn't support is so long for native app context. WebView should work.
https://github.com/appium/appium-xcuitest-driver/blob/58bf9f88ef7c119e8dc9025fcc454e3a435f5c6c/lib/commands/web.js#L199

**Comment by Dor-bl** [comment]
@KazuCocoa, Is it necessary to have a chrome driver part of appium?
Getting this errror:
Call to 'switchContext' failed
An unknown server-side error occurred while processing the command. Original error: No Chromedriver found that can automate Chrome '127.0.6533'.
![image](https://github.com/user-attachments/assets/707c4f4e-2a68-41e4-8061-4b32c260c5e1)

**Comment by KazuCocoa** [comment]
Usually yes if your local Appium does not have the proper version of chromedriver. Auto download will help. https://github.com/appium/appium-uiautomator2-driver?tab=readme-ov-file#automatic-discovery-of-compatible-chromedriver

**Comment by KazuCocoa** [comment]
This is with Safari for xcuitest driver, just for a note the method itself hasn't been deprecated in WEBVIEW context.
<img width="1092" alt="Screenshot 2024-09-03 at 11 39 51 PM" src="https://github.com/user-attachments/assets/99f6fff1-effd-4cbb-9995-0679e277c42c">

**Comment by Dor-bl** [comment]
not sure how, but on my real device no WEBVIEW contexts are found for iOS
using xcuitest@5.16.1
![image](https://github.com/user-attachments/assets/ca3eaa69-5cdd-4e2e-bd9f-b125808ad943)

**Comment by Dor-bl** [comment]
I assume I need this: https://github.com/appium/appium-safari-driver?tab=readme-ov-file#appium-safari-driver :)

**Comment by KazuCocoa** [comment]
Have you configured:

https://appium.github.io/appium-xcuitest-driver/latest/preparation/#webview-testing/

and

https://appium.github.io/appium-xcuitest-driver/latest/preparation/real-device-config/

> Webviews will not be testable unless the Safari Inspector is enabled. Please turn it on in Settings -> Safari -> Advanced -> Web Inspector. Similarly, you may want to turn on the adjacent option Settings -> Safari -> Advanced -> Remote Automation.

?

**Comment by KazuCocoa** [comment]
Closed as this error was expected as non-web view context. Can keep posting comments here

**Comment by Dor-bl** [comment]
> Have you configured:
> 
> https://appium.github.io/appium-xcuitest-driver/latest/preparation/#webview-testing/
> 
> and
> 
> https://appium.github.io/appium-xcuitest-driver/latest/preparation/real-device-config/
> 
> > Webviews will not be testable unless the Safari Inspector is enabled. Please turn it on in Settings -> Safari -> Advanced -> Web Inspector. Similarly, you may want to turn on the adjacent option Settings -> Safari -> Advanced -> Remote Automation.
> 
> ?

Thanks, @KazuCocoa. Enabling those settings was enough.
So I don't really need the Safari Driver
